src/main/java/com/google/devtools/build/lib/bazel/bzlmod/RegistryFactoryImpl.java

```diff
@@ -0,0 +1,32 @@
+package com.google.devtools.build.lib.bazel.bzlmod;
+
+import com.google.devtools.build.lib.bazel.repository.downloader.HttpDownloader;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Map;
+import java.util.function.Supplier;
+
+public class RegistryFactoryImpl implements RegistryFactory {
```

Comment:
nit: `ProductionRegistryFactory`? (`FactoryImpl` is getting close to the infamous Java "FactorySupllierBuilderCacheProviderFactory" antipattern...)